src/os.cc

```diff
@@ -353,6 +354,20 @@ buffered_file file::fdopen(const char* mode) {
   return bf;
 }
 
+#  if defined(_WIN32) && !defined(__MINGW32__)
+file file::open_windows_file(wcstring_view path, int oflag) {
+  int fd_ = -1;
+  using mode_t = int;
+  constexpr mode_t mode =
+      S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH;
+  FMT_POSIX_CALL(wsopen_s(&fd_, path.c_str(), oflag, _SH_DENYNO, mode));
```

Comment:
Looks like wsopen_s returns an error code so I suggest using it instead of global errno.